ui/components/data-sources/DataSourcesContainer.js

```diff
@@ -44,7 +44,12 @@ class DataSourcesContainer extends Component {
                     }}
                 />
                 <div>
-                    <DataSourcesList filter={this.state.filter} /> {/*eslint-disable-line*/}
+                    <DataSourcesList
+                        filter={filter}
+                        onCreate={() => {
```

Comment:
onCreate={this.addDataSource}

Reply:
Unfortunately it won't work then because it needs to be called in the context of DataSourcesContainer.